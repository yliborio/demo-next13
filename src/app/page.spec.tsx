import { render, screen, waitFor, act } from "@testing-library/react";
import Home from "./page";

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () =>
      Promise.resolve([
        {
          id: 1,
          title: "Product 1",
          image: "/test-image",
          rating: { rate: 4, count: 4 },
        },
        {
          id: 2,
          title: "Product 2",
          image: "/test-image",
          rating: { rate: 4, count: 4 },
        },
      ]),
  } as any)
);

describe("Home component", () => {
  it("fetches and displays products", async () => {
    const page = await Home();
    const { asFragment } = render(page);

    expect(asFragment()).toMatchSnapshot();

    await waitFor(() => {
      expect(fetch).toHaveBeenCalledWith("https://fakestoreapi.com/products");
    });

    const product1Element = screen.getByText("Product 1 - 4/5 (4)");
    const product2Element = screen.getByText("Product 2 - 4/5 (4)");
    expect(product1Element).toBeInTheDocument();
    expect(product2Element).toBeInTheDocument();
  });
});
