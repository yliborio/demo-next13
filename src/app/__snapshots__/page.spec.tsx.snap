// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component fetches and displays products 1`] = `
<DocumentFragment>
  <main
    class="main"
  >
    <div
      class="container"
    >
      <div
        class="orderby"
        role="button"
        tabindex="0"
      >
        <span>
          Order by
        </span>
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="12"
            x2="12"
            y1="5"
            y2="19"
          />
          <polyline
            points="19 12 12 19 5 12"
          />
        </svg>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="products-grid"
      >
        <a
          class="container"
          href="/product?id=1"
        >
          <div
            class="card"
          >
            <div
              class="product-image"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                sizes="100vw"
                src="/_next/image?url=%2Ftest-image&w=3840&q=75"
                srcset="/_next/image?url=%2Ftest-image&w=640&q=75 640w, /_next/image?url=%2Ftest-image&w=750&q=75 750w, /_next/image?url=%2Ftest-image&w=828&q=75 828w, /_next/image?url=%2Ftest-image&w=1080&q=75 1080w, /_next/image?url=%2Ftest-image&w=1200&q=75 1200w, /_next/image?url=%2Ftest-image&w=1920&q=75 1920w, /_next/image?url=%2Ftest-image&w=2048&q=75 2048w, /_next/image?url=%2Ftest-image&w=3840&q=75 3840w"
                style="color: transparent;"
                width="0"
              />
            </div>
            <div
              class="title"
            >
              <span>
                Product 1 - 4/5  (4)
              </span>
            </div>
            <span
              class="divider"
            />
            <span
              class="price"
            >
              $undefined 
            </span>
          </div>
        </a>
        <a
          class="container"
          href="/product?id=2"
        >
          <div
            class="card"
          >
            <div
              class="product-image"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                sizes="100vw"
                src="/_next/image?url=%2Ftest-image&w=3840&q=75"
                srcset="/_next/image?url=%2Ftest-image&w=640&q=75 640w, /_next/image?url=%2Ftest-image&w=750&q=75 750w, /_next/image?url=%2Ftest-image&w=828&q=75 828w, /_next/image?url=%2Ftest-image&w=1080&q=75 1080w, /_next/image?url=%2Ftest-image&w=1200&q=75 1200w, /_next/image?url=%2Ftest-image&w=1920&q=75 1920w, /_next/image?url=%2Ftest-image&w=2048&q=75 2048w, /_next/image?url=%2Ftest-image&w=3840&q=75 3840w"
                style="color: transparent;"
                width="0"
              />
            </div>
            <div
              class="title"
            >
              <span>
                Product 2 - 4/5  (4)
              </span>
            </div>
            <span
              class="divider"
            />
            <span
              class="price"
            >
              $undefined 
            </span>
          </div>
        </a>
      </div>
      <div
        class="selector"
      >
        <button
          class="actions"
          data-testid="back-button"
          disabled=""
        >
          &lt;
        </button>
        <button
          class="pages"
          data-testid="page-0-button"
          disabled=""
          tabindex="0"
        >
          1
        </button>
        <button
          class="actions"
          data-testid="next-button"
          disabled=""
        >
          &gt;
        </button>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
